huaweicloud/services/ccm/resource_huaweicloud_ccm_certificate_push.go

```diff
@@ -0,0 +1,383 @@
+	bodyParams := map[string]interface{}{
+		"target_service":  d.Get("service"),
+		"target_projects": buildTargetProject(d.Get("targets").(*schema.Set).List()),
+	}
+	return bodyParams
+}
+
+func buildUpdatePushCertificateBodyParams(service string, projectlist []interface{}) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"target_service":  service,
+		"target_projects": buildTargetProject(projectlist),
+	}
+	return bodyParams
+}
+func buildTargetProject(projectlist []interface{}) []string {
+	var projects []string
+	for _, v := range projectlist {
+		projects = append(projects, utils.PathSearch("project_name", v, nil).(string))
+	}
+	return projects
+}
+
+func flattenResult(resp interface{}) []interface{} {
+	curJson := utils.PathSearch("results", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		rst = append(rst, map[string]interface{}{
+			"project_name": utils.PathSearch("project_name", v, nil),
+			"cert_id":      utils.PathSearch("cert_id", v, nil),
+		})
+	}
+	return rst
+}
+
+func flattenUpdateResult(d *schema.ResourceData, resp interface{}) []interface{} {
+	targets := d.Get("targets").(*schema.Set).List()
+	rst := make([]interface{}, 0, len(targets))
+	for _, v := range targets {
+		certId := utils.PathSearch("cert_id", v, nil).(string)
+		projectName := utils.PathSearch("project_name", v, nil).(string)
+		if certId == "" {
+			certId = getCertIdFromResult(projectName, resp)
+		}
+		rst = append(rst, map[string]interface{}{
+			"project_name": projectName,
+			"cert_id":      certId,
+			"cert_name":    utils.PathSearch("cert_name", v, nil),
+		})
+	}
+
+	return rst
+}
+
+func getCertIdFromResult(projectName string, resp interface{}) string {
+	var certId = ""
+	curJson := utils.PathSearch("results", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	for _, v := range curArray {
+		resName := utils.PathSearch("project_name", v, nil).(string)
+		if projectName == resName {
+			certId = utils.PathSearch("cert_id", v, nil).(string)
+			break
+		}
+	}
+	return certId
+}
+func resourceCcmCertificateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	projectList := d.Get("targets").(*schema.Set).List()
+	targetService := d.Get("service").(string)
+	rst := make([]interface{}, 0, len(projectList))
+	if targetService == targetServiceElb {
+		for _, val := range projectList {
+			projectName := utils.PathSearch("project_name", val, nil).(string)
+			certId := utils.PathSearch("cert_id", val, nil).(string)
+			cfg := meta.(*config.Config)
+			elbClient, err := cfg.ElbV3Client(projectName)
+			if err != nil {
+				return diag.Errorf("error creating ELB client: %s", err)
+			}
+
+			cert, err := certificates.Get(elbClient, certId).Extract()
+			if err != nil {
+				return common.CheckDeletedDiag(d, err, "error retrieving ELB certificate")
```

Comment:
just add the log rather than report an error.

Reply:
done